server/views.py

```diff
@@ -18,15 +19,18 @@ def get_user_info_dict(user):
 
 
 def index(request):
+    notebooks = Notebook.objects \
+        .annotate(latest_revision=Max('revisions__created')) \
+        .order_by('-latest_revision') \
+        .values_list('id', 'title', 'owner__username', 'owner__avatar')
     return render(
         request, 'index.html', {
             'page_data': {
                 'userInfo': get_user_info_dict(request.user),
                 # this is horrible and will not scale
                 'notebookList': [
                     {'id': v[0], 'title': v[1], 'owner': v[2], 'avatar': v[3]}
-                    for v in
-                    Notebook.objects.values_list('id', 'title', 'owner__username', 'owner__avatar')
+                    for v in notebooks
```

Comment:
For clarity, I think we should be able to rewrite this as:

```py
'notebookList': [
    { 'id': id, 'title': title, 'owner': owner, 'avatar': avatar } for (id, title, owner, avatar) in notebooks
]
```

Reply:
@wlach I'm going to go ahead and change this in `index` as well which is where I copied from